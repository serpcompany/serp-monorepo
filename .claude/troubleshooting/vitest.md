https://vitest.dev/guide/common-errors.html