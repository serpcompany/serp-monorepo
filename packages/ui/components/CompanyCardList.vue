<script setup lang="ts">
  import type { Company } from '@serp/types/types';

  defineProps({
    companies: {
      type: Array as PropType<Company[]>,
      required: true
    },
    showReadMore: {
      type: Boolean,
      default: false
    },
    showProsAndCons: {
      type: Boolean,
      default: false
    },
    showFeatures: {
      type: Boolean,
      default: false
    },
    showExpandedContent: {
      type: Boolean,
      default: false
    }
  });
</script>

<template>
  <div v-for="company in companies" :key="company.id" class="mb-4 border p-4">
    <CompanyCard
      :company="company"
      :show-read-more="showReadMore"
      :show-pros-and-cons="showProsAndCons"
      :show-features="showFeatures"
      :show-expanded-content="showExpandedContent"
    />
  </div>
</template>
