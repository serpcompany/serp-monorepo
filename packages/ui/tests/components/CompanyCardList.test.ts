import { describe, expect, it } from 'vitest';
import { defineComponent } from 'vue';
import CompanyCardList from '../../components/CompanyCardList.vue';
import ComponentRender from '../componentRender';

// Minimal stub for the external CompanyCard dependency
const CompanyCardStub = defineComponent({
  props: [
    'company',
    'showReadMore',
    'showProsAndCons',
    'showFeatures',
    'showExpandedContent',
  ],
  template: '<div class="company-card-stub">{{ company.name }}</div>',
});

describe('companyCardList Snapshot', () => {
  const scenarios = [
    [
      'with an empty companies list',
      {
        props: { items: [] },
        global: { stubs: { CompanyCard: CompanyCardStub } },
      },
    ],
    [
      'with a single company and default props',
      {
        props: { items: [{ id: 1, name: 'Test Company' }] },
        global: { stubs: { CompanyCard: CompanyCardStub } },
      },
    ],
    [
      'with multiple companies and all toggles enabled',
      {
        props: {
          items: [
            { id: 1, name: 'Company One' },
            { id: 2, name: 'Company Two' },
          ],
          showReadMore: true,
          showProsAndCons: true,
          showFeatures: true,
          showExpandedContent: true,
        },
        global: { stubs: { CompanyCard: CompanyCardStub } },
      },
    ],
  ];

  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, options: { props: unknown; global?: unknown }) => {
      const html = await ComponentRender(
        `CompanyCardList ${desc}`,
        options,
        CompanyCardList,
      );
      expect(html).toMatchSnapshot();
    },
  );
});
