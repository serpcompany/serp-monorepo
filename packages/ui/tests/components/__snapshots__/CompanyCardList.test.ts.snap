// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyCardList Snapshot > renders with a single company and default props correctly 1`] = `
"<div class="mb-4 border p-4">
  <div class="company-card-stub">Test Company</div>
</div>"
`;

exports[`CompanyCardList Snapshot > renders with an empty companies list correctly 1`] = `""`;

exports[`CompanyCardList Snapshot > renders with multiple companies and all toggles enabled correctly 1`] = `
"<div class="mb-4 border p-4">
  <div class="company-card-stub">Company One</div>
</div>
<div class="mb-4 border p-4">
  <div class="company-card-stub">Company Two</div>
</div>"
`;
